crates/bin/c8y_configuration_poc/src/c8y_http_proxy/actor.rs

```diff
@@ -0,0 +1,420 @@
+use crate::c8y_http_proxy::credentials::JwtRequest;
+use crate::c8y_http_proxy::credentials::JwtResult;
+use crate::c8y_http_proxy::credentials::JwtRetriever;
+use crate::c8y_http_proxy::messages::C8YRestError;
+use crate::c8y_http_proxy::messages::C8YRestRequest;
+use crate::c8y_http_proxy::messages::C8YRestResult;
+use crate::c8y_http_proxy::messages::DownloadFile;
+use crate::c8y_http_proxy::messages::EventId;
+use crate::c8y_http_proxy::messages::Unit;
+use crate::c8y_http_proxy::messages::UploadConfigFile;
+use crate::c8y_http_proxy::messages::UploadLogBinary;
+use crate::C8YHttpConfig;
+use async_trait::async_trait;
+use c8y_api::http_proxy::C8yEndPoint;
+use c8y_api::json_c8y::C8yCreateEvent;
+use c8y_api::json_c8y::C8yEventResponse;
+use c8y_api::json_c8y::C8yManagedObject;
+use c8y_api::json_c8y::C8yUpdateSoftwareListResponse;
+use c8y_api::json_c8y::InternalIdResponse;
+use c8y_api::smartrest::error::SMCumulocityMapperError;
+use c8y_api::OffsetDateTime;
+use hyper::body;
+use log::error;
+use log::info;
+use serde::de::DeserializeOwned;
+use std::collections::HashMap;
+use std::time::Duration;
+use tedge_actors::fan_in_message_type;
+use tedge_actors::Actor;
+use tedge_actors::ChannelError;
+use tedge_actors::MessageBox;
+use tedge_actors::ServiceMessageBox;
+use tedge_http_ext::HttpHandle;
+use tedge_http_ext::HttpRequest;
+use tedge_http_ext::HttpRequestBuilder;
+use tedge_http_ext::HttpResult;
+
+const RETRY_TIMEOUT_SECS: u64 = 60;
+
+pub(crate) struct C8YHttpProxyActor {
+    end_point: C8yEndPoint,
+    child_devices: HashMap<String, String>,
+}
+
+#[async_trait]
+impl Actor for C8YHttpProxyActor {
+    type MessageBox = C8YHttpProxyMessageBox;
+
+    fn name(&self) -> &str {
+        "C8YHttpProxy"
+    }
+
+    async fn run(self, messages: Self::MessageBox) -> Result<(), ChannelError> {
+        C8YHttpProxyActor::run(self, messages).await
+    }
+}
+
+pub(crate) struct C8YHttpProxyMessageBox {
+    /// Connection to the clients
+    pub(crate) clients: ServiceMessageBox<C8YRestRequest, C8YRestResult>,
+
+    /// Connection to an HTTP actor
+    pub(crate) http: HttpHandle,
+
+    /// Connection to a JWT token retriever
+    pub(crate) jwt: JwtRetriever,
+}
```

Comment:
> The main drawback is that one needs to impl MessageBox for TheWholeBox. It's straightforward but tedious - one have notably to come with names for global messages (e.g C8YHttpProxyInput). Can this be derived by a macro?

This will be less a concern if the `send` and `recv` methods are deprecated from the MessageBox trait.